src/Typechecker/Typechecker.hs

```diff
@@ -44,56 +79,78 @@ deftype :: (ValDef SourcePos) -> Typechecked Type
 deftype (Val (Sig n _t) eqn x) = do
   setPos x
   eqt <- localEnv ((n, _t):) (eqntype n _t eqn)
-  if eqt <= _t
```

Comment:
`<=` to monadic `<:` is because subtyping now requires type definitions, which we store in the type checking monad.